tinygrad/engine/realize.py

```diff
@@ -186,26 +187,28 @@ def run(self, var_vals:Optional[Dict[Variable, int]]=None, wait=False, jit=False
       self.prg.first_run = False
     return et
 
-def lower_schedule_item(si:ScheduleItem) -> ExecItem:
+def lower_schedule_item(si:ScheduleItem) -> List[ExecItem]:
   assert len(set(x.device for x in si.bufs)) == 1 or (si.ast.op is UOps.EXT and si.ast.arg[0] is MetaOps.COPY)
   if si.ast.op is UOps.SINK:
-    runner = get_runner(si.outputs[0].device, si.ast)
-    return ExecItem(runner, [si.bufs[x] for x in runner.p.globals], si.metadata)
+    dname = si.outputs[0].device
+    runners = [get_runner(dname, k) for k in get_kernels(Device[dname].renderer, si.ast)]
```

Comment:
are you recreating Kernel for si.ast every time?

Reply:
As much as before and the same cache. Just had to move `get_kernel` out of `get_runner` because `get_kernels` could return multiple Kernels after `OptOps.SPLIT`.

Reply:
I see let me fix that..

Reply:
Should be caching again

Reply:
Can you add a test for it in test_method_cache?